### ISSUE TITLE
Instantiation fails with TypeError only when fail_untyped=False

### ISSUE THREAD

**Issue by cristianregep** [omni-us/jsonargparse#362]
## 🐛 Bug report

I'm trying to instantiate a config class from Huggingface with CLI and it fails when `fail_untyped=False`. The fail_untyped is important because I'm doing this as part of a project that is written within the lightning framework, where we make heavy use of lightning cli, which sets fail_untyped to False and you can't change it. Most of our project is written internally by us and is not HuggingFace based, but we want to make use from time to time of some external models in the open source community (like for example HuggingFace). 

I created an isolated example just with jsonargparse (see below). It only fails when `fail_untyped=False` . I stepped through the instantiation and I'm quite confused of what's happening. `num_labels` is not set, and through kwargs.pop("num_labels", 2) it should become 2, which is an int.

```
File "/opt/pyenv/versions/3.10.12/lib/python3.10/site-packages/transformers/configuration_utils.py", line 331, in __init__
    self.num_labels = kwargs.pop("num_labels", 2)

File "/opt/pyenv/versions/3.10.12/lib/python3.10/site-packages/transformers/configuration_utils.py", line 256, in __setattr__
    super().__setattr__(key, value)

File "/opt/pyenv/versions/3.10.12/lib/python3.10/site-packages/transformers/configuration_utils.py", line 420, in num_labels
    self.id2label = {i: f"LABEL_{i}" for i in range(num_labels)}

TypeError: 'NoneType' object cannot be interpreted as an integer
```

As a side note I am planning to submit a PR soon to allow people to set fail_untyped in lightning https://github.com/Lightning-AI/lightning/issues/18285  . 

### To reproduce
```python
from jsonargparse import CLI
from dataclasses import dataclass
from transformers.models.perceiver.configuration_perceiver import PerceiverConfig


@dataclass
class Test:
    cfg: PerceiverConfig


print(CLI(Test, fail_untyped=False))
```

Config
```yaml
cfg:
  class_path: transformers.models.perceiver.configuration_perceiver.PerceiverConfig 
```

Command
```
python perceiver.py --config perceiver_config.yaml
``` 

### Expected behavior

No TypeError

### Environment

- jsonargparse version:  4.23.1
- Python version: 3.10.12
- How jsonargparse was installed: `pip install jsonargparse[signatures,typing-extensions]==4.23.1`
- Huggingface transformers package: transformers==4.31.0
- OS: Ubuntu 22.04.2 LTS

**Comment by mauvilsa** [comment]
Thank you for reporting!

From what I see, the conditional defaults from the AST resolver is not working correctly for the case of `kwargs.pop()` (i.e. [configuration_utils.py#L322-L331](https://github.com/huggingface/transformers/blob/07998ef39926b76d3f6667025535d0859eed61c3/src/transformers/configuration_utils.py#L322-L331)). I will fix it.

For more details about the expected behavior, search for `conditional_calls(**kwargs)` in the [docs](https://jsonargparse.readthedocs.io/en/v4.24.0/#ast-resolver).

**Comment by mauvilsa** [comment]
I implemented a fix in #364. @cristianregep please test it out.

**Comment by cristianregep** [comment]
Thanks for the quick turnaround. It works
